gnu-sort/cmd/main_test.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"bufio"
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestReadLines(t *testing.T) {
+	// Prepare a test file
+	testFile := "test_input.txt"
+	lines := []string{"apple", "banana", "cherry"}
+	file, _ := os.Create(testFile)
+	writer := bufio.NewWriter(file)
+	for _, line := range lines {
+		writer.WriteString(line + "\n")
+	}
+	writer.Flush()
+	file.Close()
+
+	// Test ReadLines function
+	readLines, err := ReadLines(testFile)
+	if err != nil {
+		t.Errorf("Failed to read lines: %s", err)
+	}
+
+	if !reflect.DeepEqual(readLines, lines) {
+		t.Errorf("ReadLines = %v; want %v", readLines, lines)
+	}
+
+	// Cleanup
+	os.Remove(testFile)
+}
+
+func TestSortLines(t *testing.T) {
```

Comment:
I would leave this as the sole test but with multiple cases. `TestReadLines` and `TestWriteLines` are not really part of the core business logic so we can leave them out. Tip: have a look at "table tests" in golang

Reply:
Focused on table tests for SortLines. Dropped tests for ReadLines and WriteLines.